File: SQLiteRepository/SQLiteQuery/QueryResult.cs
using System.Collections.Generic;
using System.Dynamic;

namespace SQLiteRepository.SQLiteQuery
{
    public class QueryResult : IQueryResult
    { 
        public int AffectedRows { get; set; }
        public List<object> ScalarResults { get; set; }
      
        public List<List<ExpandoObject>> TabularResults { get; set; }
        public List<string> SqlCommands { get; set; }

    }
}

File: SQLiteAdapters/QueryResult.vb
Imports System.Collections.Generic
Imports System.Dynamic


Public Class QueryResult
    Implements IQueryResult
    Public Property AffectedRows As Integer Implements IQueryResult.AffectedRows

    Public Property ScalarResults As List(Of Object) Implements IQueryResult.ScalarResults

    Public Property TabularResults As List(Of List(Of ExpandoObject)) Implements IQueryResult.TabularResults

    Public Property SqlCommands As List(Of String) Implements IQueryResult.SqlCommands
End Class